implementations/java-jsonschema-validator/bowtieJsonSchemaValidator.java

```diff
@@ -0,0 +1,215 @@
+  private final JsonSchemaFactory schemaFactory =
+      JsonSchemaFactory.getInstance();
+  private final PrintStream output;
+
+  public static void main(String[] args) {
+    BufferedReader reader =
+        new BufferedReader(new InputStreamReader(System.in));
+    new BowtieJsonSchemaValidator(System.out).run(reader);
+  }
+
+  public BowtieJsonSchemaValidator(PrintStream output) { this.output = output; }
+
+  private void run(BufferedReader reader) {
+    reader.lines().forEach(this::handle);
+  }
+
+  private void handle(String data) {
+    try {
+      JsonNode node = objectMapper.readTree(data);
+      String cmd = node.get("cmd").asText();
+      switch (cmd) {
+        case "start" -> start(node);
+        case "dialect" -> dialect(node);
+        case "run" -> run(node);
+        case "stop" -> System.exit(0);
+        default -> throw new IllegalArgumentException(
+          "Unknown cmd [%s]".formatted(cmd)
+        );
+      }
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private void start(JsonNode node) throws IOException {
+    StartRequest startRequest = objectMapper.treeToValue(
+      node,
+      StartRequest.class
+    );
+    if (startRequest.version() != 1) {
+      throw new IllegalArgumentException(
+        "Unsupported IHOP version [%d]".formatted(startRequest.version())
+      );
+    }
+
+    InputStream is = getClass().getResourceAsStream("META-INF/MANIFEST.MF");
+    var attributes = new Manifest(is).getMainAttributes();
+
+    StartResponse startResponse = new StartResponse(
+      1,
+      true,
+      new Implementation(
+        "java",
+        attributes.getValue("Implementation-Name"),
+        attributes.getValue("Implementation-Version"),
+        DIALECTS,
+        "https://github.com/networknt/json-schema-validator/",
+        "https://github.com/networknt/json-schema-validator/issues",
+        System.getProperty("os.name"),
+        System.getProperty("os.version"),
+        System.getProperty("java.vendor.version")
+      )
+    );
+    output.println(objectMapper.writeValueAsString(startResponse));
+  }
+
+  private void dialect(JsonNode node) throws JsonProcessingException {
+    DialectRequest dialectRequest = objectMapper.treeToValue(
+      node,
+      DialectRequest.class
+    );
+    if (DIALECTS.contains(dialectRequest.dialect())) {
+      output.println(
+        objectMapper.writeValueAsString(new DialectResponse(true))
+      );
+    } else {
+      output.println(
+        objectMapper.writeValueAsString(new DialectResponse(false))
+      );
+    }
+  }
+
+  private void run(JsonNode node) throws JsonProcessingException {
+    RunRequest runRequest = objectMapper.treeToValue(node, RunRequest.class);
+    try {
+      if (runRequest.testCase().registry() != null) {
+        JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance();
+        JsonSchema schema = schemaFactory.getSchema(
```

Comment:
Does it do something? It should somehow register all schema from `runRequest.testCase().registry()`, but unfortunately I don't know the API of networknt lib

Reply:
Yes, the `getSchema` method is responsible for collecting the schema that needs to be validated. [This ](https://github.com/networknt/json-schema-validator/blob/master/doc/quickstart.md)is the reference I have used.

Reply:
Yes, the `getSchema` method is responsible for collecting the schema that needs to be validated. [This ](https://github.com/networknt/json-schema-validator/blob/master/doc/quickstart.md)is the reference I have used to for this implementation.

Reply:
Thank you for taking a close look at the PR! It's quite late today, so I will address your comments and resolve/respond to them tomorrow.

Reply:
Sure thing, there is no rush

Reply:
So, if I recall correctly the structure returned from `runRequest.testCase().registry()` is something like this:
```json
{
  "https://host.com/schema1": { ...here goes actual schema definition },
  "https://host.com/schema2": { ...here goes actual schema definition },
  ...
}
```
so you would need to iterate over all fields in this JSON and call this `getSchema()` with value of each field.

But I have other concern that `getSchema` only creates Schema object and it does not store it somewhere in its internal registry (but I might be wrong, haven't had time to thoroughly check it). Maybe implementing custom fetcher is a way to go? https://github.com/networknt/json-schema-validator/blob/master/doc/cust-fetcher.md
In this fetcher you would just serve schemas from beforementioned registry json object

Reply:
used ArrayList to store the results